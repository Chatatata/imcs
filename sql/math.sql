select cs_sqrt((Open - Close) ^ 2.0) from Quote_get('IBM');
select cs_sin(Open)*cs_sin(Open) + cs_cos(Open)*cs_cos(Open) from Quote_get('IBM');
select cs_atan(cs_tan(Close)) from Quote_get('IBM');
select cs_asin(cs_sin(Close)) from Quote_get('IBM');
select cs_acos(cs_cos(Close)) from Quote_get('IBM');
select cs_log(cs_exp(Close/Open)) - Open/Close from Quote_get('IBM');
select cs_isnan(cs_parse('{-1,0,1}','float8')/0.0);