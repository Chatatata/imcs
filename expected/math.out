select cs_sqrt((Open - Close) ^ 2.0) from Quote_get('IBM');
                  cs_sqrt                  
-------------------------------------------
 float8:{0.3,0,0.299999,0.299999,0.299999}
(1 row)

select cs_sin(Open)*cs_sin(Open) + cs_cos(Open)*cs_cos(Open) from Quote_get('IBM');
      ?column?      
--------------------
 float8:{1,1,1,1,1}
(1 row)

select cs_atan(cs_tan(Close)) from Quote_get('IBM');
                       cs_atan                        
------------------------------------------------------
 float8:{1.07522,1.35044,-1.21593,-0.640704,0.234518}
(1 row)

select cs_asin(cs_sin(Close)) from Quote_get('IBM');
                       cs_asin                        
------------------------------------------------------
 float8:{-1.07522,1.35044,-1.21593,0.640704,0.234518}
(1 row)

select cs_acos(cs_cos(Close)) from Quote_get('IBM');
                      cs_acos                      
---------------------------------------------------
 float8:{2.06637,1.35044,1.21593,2.50089,0.234518}
(1 row)

select cs_log(cs_exp(Close/Open)) - Open/Close from Quote_get('IBM');
                      ?column?                      
----------------------------------------------------
 float8:{0.0579832,0,-0.0197697,-0.01487,0.0119166}
(1 row)

select cs_isnan(cs_parse('{-1,0,1}','float8')/0.0);
   cs_isnan   
--------------
 char:{0,1,0}
(1 row)

